### PR TITLE
Jonmv/more doc api parameters

### PR DESCRIPTION
**Pull request by jonmv**
@bratseth please review. 

I realised after doing all this that the interface `MessageBusSession` lets you set route and trace level, and the `MessageBusAsyncSession` obviously implements this. If the `DocumentAccess` is supposed to be the "starting point of the document API", however, then this is not reachable. You end up having to use the concrete implementation. Setting route and trace level on the whole session is also less flexible, and requires in practice a separate session whenever any of these are to be set — or a session per route/tracelevel. 
To make fieldSet, route and traceLevel available to the users of the document API, I think they should be parameters to each operation, and reflected in the `Response`. This PR does that, but it's quite a bit of code to achieve this, cluttering APIs somewhat — in particular the `Response` hierarchy. 

Soooo ... PTAL :)

### CODE REVIEW

File: documentapi/src/main/java/com/yahoo/documentapi/AsyncSession.java [hunk 1]
```diff
@@ -158,6 +194,23 @@ default Result get(DocumentId id, DocumentProtocol.Priority priority) {
...
      */
     default Result remove(DocumentId id, DocumentProtocol.Priority priority) {
+        return remove(id, parameters().withPriority(priority));
+    }
+
+    /**
+     * <p>Removes a document if it is present. This method returns immediately.</p>
+     *
+     * <p>If this result is a success, this
+     * call will cause one or more {@link DocumentIdResponse} objects to apprear within the timeout time of this session.
```
Comment (bratseth) [inline documentapi/src/main/java/com/yahoo/documentapi/AsyncSession.java:204]:
```suggestion
     * call will cause one or more {@link DocumentIdResponse} objects to appear within the timeout time of this session.
```

File: documentapi/src/main/java/com/yahoo/documentapi/DocumentOperationParameters.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import com.yahoo.document.fieldset.FieldSet;
+import com.yahoo.document.fieldset.FieldSetRepo;
+import com.yahoo.documentapi.messagebus.protocol.DocumentProtocol;
+
+import java.util.Optional;
+import java.util.OptionalInt;
+
+import static java.util.Objects.requireNonNull;
+
+/** Optional parameters for a document operation. Immutable class. */
```
Comment (bratseth) [inline documentapi/src/main/java/com/yahoo/documentapi/DocumentOperationParameters.java:12]:
```suggestion
/** 
 * Optional parameters for a document operation. Immutable class.
 *
 * @author jonmv 
 */
```

File: documentapi/src/main/java/com/yahoo/documentapi/DocumentOperationParameters.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
+package com.yahoo.documentapi;
```
Comment (bratseth) [inline documentapi/src/main/java/com/yahoo/documentapi/DocumentOperationParameters.java:1]:
```suggestion
// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
package com.yahoo.documentapi;
```

**Review by bratseth — APPROVED** [review]
Great!

**Comment by jonmv** [comment]
Thanks!

**Pull request merged**